packages/purpose-process/src/services/purposeService.ts

```diff
@@ -317,3 +377,88 @@ const replacePurposeVersion = (
     versions: updatedVersions,
   };
 };
+
+const getInvolvedTenantByEServiceMode = async (
+  eservice: EService,
+  consumerId: TenantId,
+  readModelService: ReadModelService
+): Promise<Tenant> => {
+  if (eservice.mode === "Deliver") {
+    return retrieveTenant(consumerId, readModelService);
+  } else {
+    return retrieveTenant(eservice.producerId, readModelService);
+  }
+};
+
+const updatePurposeInternal = async (
+  purposeId: PurposeId,
+  updateContent: ApiPurposeUpdateContent | ApiReversePurposeUpdateContent,
+  organizationId: TenantId,
+  eserviceMode: EServiceMode,
+  {
+    readModelService,
+    correlationId,
+    repository,
+  }: {
+    readModelService: ReadModelService;
+    correlationId: string;
+    repository: {
+      createEvent: (createEvent: CreateEvent<PurposeEvent>) => Promise<string>;
+    };
+  }
+): Promise<{ purpose: Purpose; isRiskAnalysisValid: boolean }> => {
+  const purpose = await retrievePurpose(purposeId, readModelService);
+  assertOrganizationIsAConsumer(organizationId, purpose.data.consumerId);
+  assertPurposeIsDraft(purpose.data);
+
+  const eservice = await retrieveEService(
+    purpose.data.eserviceId,
+    readModelService
+  );
+  assertEserviceHasSpecificMode(eservice, eserviceMode);
+  assertConsistentFreeOfCharge(
+    updateContent.isFreeOfCharge,
+    updateContent.freeOfChargeReason
+  );
+
+  const tenant = await getInvolvedTenantByEServiceMode(
+    eservice,
+    purpose.data.consumerId,
+    readModelService
+  );
+
+  assertTenantKindExists(tenant);
+
+  const newRiskAnalysis: PurposeRiskAnalysisForm | undefined =
+    eserviceMode === "Deliver"
+      ? validateAndTransformRiskAnalysis(
+          (updateContent as ApiPurposeUpdateContent).riskAnalysisForm,
+          tenant.kind
+        )
+      : reverseValidateAndTransformRiskAnalysis(
+          purpose.data.riskAnalysisForm,
+          tenant.kind
+        );
+
+  const updatedPurpose: Purpose = {
+    ...purpose.data,
+    ...updateContent,
+    riskAnalysisForm: newRiskAnalysis,
+  };
```

Comment:
I think here we should add `updatedAt: new Date()`

Reply:
You're right, updated!